### PR TITLE
Add driver version to gpu info

### PR DESCRIPTION
**Pull request by satyaog**
Add driver version to gpu info.

This is to at least keep a trace of the driver version across executions of milabench

### CODE REVIEW

File: voir/instruments/gpu/hpu.py [hunk 1]
```diff
@@ -101,6 +103,11 @@ def make_gpu_info(gid, handle, selection):
         ),
         "power": fix_num(safecall(pyhlml.hlmlDeviceGetPowerUsage, handle)) / 1000.0,
         "selection_variable": "HABANA_VISIBLE_MODULES",
+        "driver": subprocess.run(
+            shlex.split("modinfo gaudi --field version"),
```
Comment (satyaog) [inline voir/instruments/gpu/hpu.py]: What would be the equivalent module name for a hpu driver?

File: voir/instruments/gpu/rocm.py [hunk 2]
```diff
@@ -205,6 +207,11 @@ def get_gpu_info(self, device):
             "temperature": temp,
             "power": power,
             "selection_variable": "ROCR_VISIBLE_DEVICES",
+            "driver": subprocess.run(
+                shlex.split("modinfo rocm --field version"),
```
Comment (satyaog) [inline voir/instruments/gpu/rocm.py]: what would be the equivalent module name for an amd/rocm driver?

File: voir/instruments/gpu/xpu.py [hunk 3]
```diff
@@ -83,6 +84,11 @@ def parse(val, type, default):
                 "temperature": parse(temp, float, 0),
                 "power": parse(power, float, 0),
                 "selection_variable": "ONEAPI_DEVICE_SELECTOR",
+                "driver": subprocess.run(
+                    shlex.split("modinfo xpu --field version"),
```
Comment (satyaog) [inline voir/instruments/gpu/xpu.py]: what would be the equivalent module name for a xpu driver?

**Comment by satyaog** [comment]
I haven't look much where to put the code. Let me know if there would be a better place to reduce the code duplication or if this works at all for non-nvidia driver

**Comment by breuleux** [comment]
I have two questions:

* Do the pynvml/rmsi/etc. APIs not provide driver version information?
* Have these been tested on the relevant platforms? I'm worried about `subprocess.run` throwing. Adding a failsafe would help.

**Comment by satyaog** [comment]
Tested only for the nvidia driver. I'll check with pynvml but I'm not sure I can find the platforms for the others and I've just thrown stuff there in hope someone would know better. I'll add failsafes

**Comment by satyaog** [comment]
Should be better now except for XPU for which I'm not sure what would go there

**Pull request merged**